core/domain/user_services_test.py

```diff
@@ -335,7 +409,51 @@ def test_get_user_settings_by_auth_id_strict_returns_user_settings(self):
     def test_get_user_settings_by_auth_id_strict_for_missing_auth_id_is_none(
             self):
         with self.assertRaisesRegex(Exception, 'User not found.'):
-            user_services.get_user_settings_by_auth_id('auth_id_x', strict=True)
+            user_services.get_user_settings_by_auth_id(
+                'auth_id_x',
+                strict=True
+        )
+
+    def test_get_users_setting_system_user_id_adds_system_user_setting(self):
+        user_id = feconf.SYSTEM_COMMITTER_ID
+        user_ids = [user_id]
+
+        roles = [
+            feconf.ROLE_ID_FULL_USER,
+            feconf.ROLE_ID_CURRICULUM_ADMIN,
+            feconf.ROLE_ID_MODERATOR,
+            feconf.ROLE_ID_VOICEOVER_ADMIN
+        ]
+
+        less_than_time = datetime.datetime.utcnow()
+
+        users_settings = user_services.get_users_settings(user_ids)
+        for settings in users_settings:
```

Comment:
This is all much too complex -- you've constructed a list and you're iterating over a list with 1 element; I don't see the need for this complexity and I don't know why you need an "if" here. There shouldn't be logic in tests.

Could the test be boiled down to its essentials? If you're expecting a list with one element as a return value, assert that the list does have one element, then name that element something and use that name subsequently.